oaknational/Oak-Web-Application
# Added statoscope as build tooling

## Description
It's currently really difficult to diagnose bundle size issues, [statoscope](https://github.com/statoscope/statoscope) is some really great tooling to help with that.

It gives us in-depth reporting of module size and real world effects of that with ms download on various connections 

A really good example is the `crypto-browserify` issue. We can see that half of the size of `./pages/teachers/specialist/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads-auth` it `crypto-browserify` alone. So that fix should have a massive effect here.

<img width="917" alt="Screenshot 2024-11-27 at 08 30 47" src="https://github.com/user-attachments/assets/232ab901-0fa0-4562-8a4e-80e38165ecf3">


## Issue(s)

Moves along [reporting of bundle size](https://www.notion.so/oaknationalacademy/Reporting-of-bundle-size-867715971d0b4a6a8b39be5a823176c6)

## How to test
In development run

```
npm run build && npm run report:open
```

Build times are comparable on my local machine, we can look at stats in CI. _Note_, there should be another ticket to add the report as a CI artefact also.


## Screenshots
The report we get as output

<img width="2168" alt="Screenshot 2024-11-27 at 08 22 46" src="https://github.com/user-attachments/assets/3190180f-14a9-4fad-b0c3-239599dc9277">
